### PR TITLE
DOCS-10898-EDIFACT-2.3-ESB

### PR DESCRIPTION
**Pull request by emileemooreVT**
Copying folders

### CODE REVIEW

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
+= EDIFACT EDI Connector 2.3 - Batch Outbound EDIFACT Messages - Mule 4
+
+For B2B and EDI messages that are not time-sensitive, such as invoices and payment remittances, batch messages instead of transmitting transactions to your trading partners individually.
```
Comment (vijayang) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: Please rephrase as below:

For EDI messages that are not time-sensitive, such as invoice and payment remittances, you can use the Batch operation to merge multiple individual messages into a single interchange and send to your trading partners.

**Review by vijayang — DISMISSED** [review]

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
+= EDIFACT EDI Connector 2.3 - Batch Outbound EDIFACT Messages - Mule 4
+
+For EDI messages that are not time-sensitive, such as invoice and payment remittances, you can use the Batch Write operation to merge multiple individual messages into a single interchange to send to your trading partners.
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: Something to think about: where it is possible, _retrain_ yourself (I am trying for myself!) to recast modal uses like "you can use" to just "use," per our style ref. and https://confluence.internal.salesforce.com/display/public/DOCTEAM/modal+verbs.

Comment (kbostrom) [reply edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]:
```suggestion
For EDI messages that are not time-sensitive, such as invoice and payment remittances, use the Batch Write operation to merge multiple individual messages into a single interchange to send to your trading partners.
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+== Batching Stages
+
+Use the Anypoint EDIFACT EDI Connector Batch Write operation to merge individual EDIFACT messages into a single transmission to send to your trading partners.
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: GLOBAL re. connector names: The official name of each is "Anypoint Connector for...," with the abbreviation being "... Connector." In Connector Land, we don't slap "Anypoint" alone in front of the product name.

Comment (kbostrom) [reply edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]:
```suggestion
Use the Anypoint Connector for EDIFACT EDI (EDIFACT EDI Connector) Batch Write operation to merge individual EDIFACT messages into a single transmission to send to your trading partners.
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Implementing a batch involves two stages:
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: Alternatively,
```suggestion
You implement a batch in two stages:
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+. Accumulate the transactions.
+. Release the batch.
+
+=== Accumulate the Transactions
+
+To accumulate transactions for a batch:
+
+. Transform the payloads into EDIFACT messages using DataWeave and EDIFACT as you receive application messages (such as a JSON invoice) from the backend systems.
+. Create EDIFACT payloads using temporary control number keys in the EDIFACT Write operation. +
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc:21]: Add a blank line, because this step includes longish additional content.

Comment (kbostrom) [reply edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc:21]:
```suggestion
. Create EDIFACT payloads using temporary control number keys in the EDIFACT Write operation. +
+
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+* Transforms to an `EDIFACT D97A ORDERS` message.
+* Generates the intermediate EDIFACT message payload using the EDIFACT Write operation.
+* Stores the EDIFACT payload in a database table.
+
+=== Release the Batch
+
+Trading partners have varying criteria for how they want transactions batched:
+
+* Release the batch at a specific time interval or a specific time of the day.
+* Release the batch when a specific number of transactions is accumulated.
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: accumulated > reached (or similar)
"a number" is not "accumulated," although "transactions" are...in which the verb becomes plural: "are accumulated."

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+* Release the batch when the total size of the accumulated transactions is near a threshold.
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: a "specified" threshold?

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+To release the batch:
+
+. Implement the logic or module to monitor the transactions accumulated in the temporary storage and determine when a particular batch is released.
+
+. Retrieve the accumulated intermediate EDIFACT messages from the temporary storage based on the batch criteria, group the transactions together, and then send the appended payload to the EDIFACT Batch Write operation. +
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: Ditto Line 19:
```suggestion
. Retrieve the accumulated intermediate EDIFACT messages from the temporary storage based on the batch criteria, group the transactions together, and then send the appended payload to the EDIFACT Batch Write operation. +
+
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+You can either use the control number key on the Batch Write operation or leave the control number key fields empty at the time of Batch Write operation execution, allowing the EDIFACT EDI Connector to generate the control number sequence automatically based on the sender and receiver identifier combination:
+`<edifact:batch doc:name="Batch" doc:id="3ce80437-63e2-404c-b875-531708b7bfdb" config-ref="EDIFACT_Batch" interchangeControlNumberKey="NTO-MYTHICAL-UNB-TMP"> <edifact:batch-content ><![CDATA[#[vars.mergedInput]]]></edifact:batch-content> </edifact:batch>`.
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: Here and at Line 20, I wonder if we should be formatting these strings as syntax examples, which sets them apart from the text for better visibility? Ask a teammate?

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+. Ensure that the input payload sent to the Batch Write operation contains EDI transactions with data elements in the UNB envelope segments, such as `Sender ID`, `Receiver ID`, and `Version number`. +
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: Per Step 21, end with colon?

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+image::edifact-edi-connector-batch-flow-2.jpg[Release the batch]
+
+For example, this implementation of a Mule flow:
+
+* Receives an HTTP request to release accumulated EDIFACT messages transactions for a specific partner. The request is triggered after the batch release criteria are met.
+* Retrieves the intermediate EDIFACT messages from the database table.
+* Applies a transformation to group the individual EDIFACT messages together.
```
Comment (kbostrom) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc]: "group together" is redundant.

**Review by kbostrom — DISMISSED** [review]
CE edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-config-topics.adoc [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+|batchSendUniqueMessageNumbers="false" |Send unique message reference numbers (UNH.1). `false` (default) assigns message reference numbers sequentially within each interchange and reuses them in a different interchange. `true` assigns unique transaction set numbers across all interchanges sent to the same partner. It is derived by the interchange sender and recipient identification.
+|batchInitialInterchangeReference="1" |The initial interchange control reference used for outgoing messages.
+|batchInitialMessageReference="1" |The initial Message Reference Number used for outgoing messages.
+|batchRequestAcks="false" |Request acknowledgments for Send interchanges using the Acknowledgment Request field (UNB.9).
+|batchTestIndicator="" |Test indicator digit used on Send interchanges (UNB.11). By default interchanges are not sent as tests.
+|batchUseSuppliedValues="false" |Choose your own data values for control-segment identifiers such as `UNB` and `UNZ` segments or use the supplied values. `false` always generates control numbers when writing letting you choose values.
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-config-topics.adoc]:
```suggestion
|batchUseSuppliedValues="false" |Choose your own data values for control-segment identifiers such as `UNB` and `UNZ` segments or use the supplied values. `false` always generates control numbers when writing, letting you choose values.
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-examples.adoc [hunk 13]
```diff
@@ -0,0 +1,124 @@
+= EDIFACT EDI Connector 2.3 - Examples - Mule 4
+
+The following examples show how to read, validate, and write EDI messages. In addition, you can send functional and interchange acknowlegments to your trading partners.
+
+== XML for EDIFACT Studio
+Paste this code into the Studio XML editor to quickly load the flow for this
+example into your Mule app.
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-examples.adoc]:
```suggestion
example into your Mule app:
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-examples.adoc [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+       </ee:message>
+     </ee:transform>
+     <edifact:write doc:name="Write" config-ref="EDIFACT_EDI_Config"/>
+  </flow>
+</mule>
+----
+
+== Read and Validate EDIFACT EDI Messages
+
+. To read an EDIFACT message, search the palette for EDIFACT EDI, and drag the EDIFACT **Read** operation into a flow.
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-examples.adoc]:
```suggestion
. To read an EDIFACT message, search the *Mule Palette* for EDIFACT EDI, and drag the EDIFACT **Read** operation into a flow.
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-examples.adoc [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+|fatal |Flag for a fatal error. This means the associated message, group, or interchange is rejected as a result of the error.
+|errorCode |EDIFACT syntax error code
+|errorText |Text description of the error
+|===
+
+Error data is returned by the Read operation as optional lists with the `Errors` key at both the root level of the data structure and at the message level. For the message level, the list contains non-fatal errors encountered during the message parsing. For the root level, the list contains both interchange errors and fatal message errors.
+
+== Write EDIFACT EDI Messages
+
+* To write an outgoing message, search the palette for EDIFACT EDI and drag the EDIFACT **Write** operation into a flow.
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-examples.adoc]:
```suggestion
* To write an outgoing message, search the *Mule Palette* for EDIFACT EDI and drag the EDIFACT **Write** operation into a flow.
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 16]
```diff
@@ -0,0 +1,253 @@
...
+* Read
+* Write
+* Write batch
+
+== Read
+
+`<edifact:read>`
+
+
+Transform an input stream of EDI text into maps and lists structure of EDI data.
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc:94]:
```suggestion
Merge individual EDIFACT messages created and accumulated over a period of time into a single interchange (UNB).
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 17]
```diff
@@ -0,0 +1,253 @@
...
+
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | |x
+| Read Content | Binary |  |  `#[payload]` |
+| Target Variable | String |  The name of a variable on which the operation's output is placed. |  |
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
| Target Variable | String |  Name of the variable that stores the operation's output. |  |
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 18]
```diff
@@ -0,0 +1,253 @@
...
+| Target Value | String |  An expression to evaluate against the operation's output and the outcome of that expression is stored in the target variable. |  `#[payload]` |
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
| Target Value | String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  `#[payload]` |
```

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc (hunk 16 above)
Comment (valkyrie69) [reply edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc:94]: ?

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 19]
```diff
@@ -0,0 +1,253 @@
...
+* EDIFACT:SCHEMA
+* EDIFACT:WRITE
+* EDIFACT:UNKNOWN
+* EDIFACT:PARSE
+
+== Write
+
+`<edifact:write>`
+
+Transform maps and lists structure of EDI data to EDI text stream.
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc]:
```suggestion
Transforms the maps and lists structure of EDI data to EDI text stream.
```

Comment (valkyrie69) [reply edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc]: ?

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 20]
```diff
@@ -0,0 +1,253 @@
...
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | |x
+| Write Content | Object |  |  `#[payload]` |
+| Streaming Strategy | repeatable-in-memory-stream
+repeatable-file-store-stream
+non-repeatable-stream |  Configure to use repeatable streams and their behavior. |  |
+| Target Variable | String |  The name of a variable on which the operation's output is placed. |  |
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc:142]: Same comment as previous. These updates can be made in a future iteration though. Not important right now.

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc [hunk 21]
```diff
@@ -0,0 +1,253 @@
...
+* EDIFACT:SCHEMA
+* EDIFACT:WRITE
+* EDIFACT:UNKNOWN
+* EDIFACT:PARSE
+
+== Write batch
+
+`<edifact:write-batch>`
+
+Transform multiple instances of maps and lists structures of EDI data stringed together into a single grouped EDI text stream.
```
Comment (valkyrie69) [inline edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc]: the maps and lists structure?
I'm not sure about this so ignore if it doesn't make sense.

**Review by valkyrie69 — DISMISSED** [review]

**Review by vijayang — DISMISSED** [review]

File: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc (hunk 21 above)
Comment (vijayang) [reply edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc]: Rephrase this as:
Merge individual EDIFACT messages created and accumulated over a period of time into a single interchange (UNB).

**Review by valkyrie69 — APPROVED** [review]
